v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -22,99 +32,190 @@ export default class RecordFactory {
         return RecordFactory.instance;
     }
 
+    updateCacheMsg() {
+        this.cacheMsgRecord = {};
+
+        const writtenAt = new Date();
+
+        const cachedFields = this.config.getCacheMsgFields();
+        let cache: any = {};
```

Comment:
Could we use a map type here instead of any?

Reply:
Using a Map would be a good idea to get make use of typing. However, there are other aspects to take into account. By using the cache as a Map we would have make use of Map accessors (e.g. `cache.set(key,value)`) and also iterate all map fields when merging the cache with the record object. This could also affect the performance.

Since the implementation is not so trivial, I would leave it for a separate ticket/branch.